ddl/db_change_test.go

```diff
@@ -349,6 +349,49 @@ type expectQuery struct {
 	rows []string
 }
 
+func (s *testStateChangeSuite) TestAppendEnum(c *C) {
```

Comment:
I think we should test the `ALTER TABLE ... MODIFY COLUMN` statements which modify the enum column to another column type.

Reply:
What's the test result you expected? An error message?

Reply:
According to the `modifiable()` function, some changes are allowed, while others are not.

Reply:
At this time, to.Tp is always be enum.  If we alter enum's column to other type, we will got the error message:  the number of enum column's elements is less than the original. 

Is that all right?

Reply:
Yes, for example:
```sql
CREATE TABLE ti1(a INT NOT NULL, b INT, c INT, e ENUM('a', 'b')) engine=InnoDB;

ALTER TABLE ti1 MODIFY COLUMN e ENUM('a', 'b', 'c'); -- succeed 
ALTER TABLE ti1 MODIFY COLUMN e ENUM('a'); -- failed
```

Reply:
Hi, I have add some new test, please check it. thank you.